File: lecture_008(ArrayListandStringBuilder)/l_002_stack.cpp
#include<iostream>
#include<stack>
using namespace std;

void solution(stack<int> &st) {
    st.push(10);
    st.push(30);
    st.push(15);
    /* cout<<"[";
    for(int i=0; i<st.size()-1; i++) {
        cout<<st[i]<<", ";                    //error
    }
    cout<<st[st.size()-1]<<"]"<<endl; */
    cout<<"[";
    while(st.size()!=1) {
        cout<<st.top()<<", ";
        st.pop();
    }
    cout<<st.top()<<"]"<<endl;

    /* while(!st.empty()) {
        cout<<st.top()<<" ";       //gives toppest value in the stack
        st.pop();           //doesn't give toppest value , it just removes toppest value
    }
    cout<<endl; */

    st.push(20);
    cout<<st.top()<<endl;
}
int main() {
    stack<int> st;
    solution(st);

    return 0;
}

File: lecture_008(ArrayListandStringBuilder)/l_002_stack.java
import java.util.Scanner;
import java.util.Stack;

public class l_002_stack {
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args) {
        Stack<Integer> st = new Stack<>();
        solution(st);
    }
    public static void solution(Stack<Integer> st) {
        st.push(10);
        st.push(30);
        st.push(15);
        System.out.println(st);
        st.push(20);
        System.out.println(st);

        System.out.println(st.peek());   //give toppest value in the stack
        System.out.println(st);

        System.out.println(st.pop());    //gives toppest value in the stack and removes it 
        System.out.println(st);
    }
}